src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -902,10 +903,14 @@ public static string Evaluate(string format, params FormatParam[] args)
                                     ~value);
                                 break;
 
-                            // Augment first two parameters by 1
+                            // Some terminfo files make liberal use of %i to mean increment only one argument, not two.
+                            // To make sure we don't break on these files, check if we have one or two variables and increment
+                            // the proper number according to how many are passed.
                             case 'i':
-                                args[0] = 1 + args[0].Int32;
-                                args[1] = 1 + args[1].Int32;
+                                if (args.Length > 0)
+                                    args[0] = 1 + args[0].Int32;
+                                if (args.Length > 1)
+                                    args[1] = 1 + args[1].Int32;
```

Comment:
I know I scribbled it out this way in email, but we could make it slightly better by doing:
```C#
if (args.Length > 0)
{
    args[0] = 1 + args[0].Int32;
    if (args.Length > 1)
        args[1] = 1 + args[1].Int32;
}